arrow-libs/optics/arrow-optics-reflect/src/main/kotlin/arrow/optics/Reflection.kt

```diff
@@ -54,13 +57,23 @@ public val <S, A> KProperty1<S, List<A>>.every: Every<S, A>
 public val <S, K, A> KProperty1<S, Map<K, A>>.values: Every<S, A>
   get() = lens compose Every.map()
 
-private fun <S, A> clone(prop: KProperty1<S, A>, value: S, newField: A): S {
+private val <S, A> KProperty1<S, A>.reflectiveCopy: (S, A) -> S get() {
   // based on https://stackoverflow.com/questions/49511098/call-data-class-copy-via-reflection
-  val klass = prop.instanceParameter?.type?.classifier as? KClass<*>
-  val copy = klass?.memberFunctions?.firstOrNull { it.name == "copy" }
-  if (klass == null || !klass.isData || copy == null) {
+  val klass = kClass
+  val copy = klass.copyMethod
+  val fieldParam = copy.parameters.first { it.name == name }
+  val instanceParam = copy.instanceParameter!!
+  return { value, newField -> klass.cast(copy.callBy(mapOf(instanceParam to value, fieldParam to newField))) }
+}
+
+@Suppress("UNCHECKED_CAST")
+private val <S> KProperty1<S, *>.kClass: KClass<S & Any>
```

Comment:
I don't understand 100% why the result here is `KClass<S & Any>`, instead of merely `KClass<*>`.